packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1725,6 +1726,32 @@ function validateAttributes(
     }
 }
 
+function validateSlotAttribute(
+    ctx: ParserCtx,
+    parsedAttr: ParsedAttribute,
+    parentNode: ParentNode,
+    element: BaseElement
+): void {
+    const slotAttr = parsedAttr.get('slot');
+
+    if (!slotAttr) {
+        return;
+    }
+
+    const elementOrSlotAncestor = ctx.findAncestor(
+        (node) => ast.isElement(node) || ast.isSlot(node),
+        ({ current }) => current && !ast.isComponent(current) && !ast.isExternalComponent(current),
+        parentNode
+    );
```

Comment:
I'm too familiar with `ctx.findAncestor` yet so this could be wrong or inefficient.

Reply:
If I'm not mistaken, there are already other tests would fail, but I added a `valid` fixture which covers this and `lwc:external` as well.